libsolidity/grammar.txt

```diff
@@ -1,48 +1,96 @@
-ContractDefinition = ( 'contract' | 'library' ) Identifier
+ContractDefinition = ('import' Imports ';')*
+                     ( 'contract' | 'library' ) Identifier
                      ( 'is' InheritanceSpecifier (',' InheritanceSpecifier )* )?
                      '{' ContractPart* '}'
-ContractPart = StateVariableDeclaration | StructDefinition | ModifierDefinition | FunctionDefinition | EnumDefinition
+
+Imports = StringLiteral ('as' Identifier)?
+        | ('*' | Identifier) ('as' Identifier)? 'from' StringLiteral
+        | '{' Identifier ('as' Identifier)? ( ',' Identifier ('as' Identifier)? )* '}' 'from' StringLiteral
+
+ContractPart = StateVariableDecOrDef | UsingDeclaration
+             | StructDefinition | ModifierDefinition | FunctionDefinition | EventDefinition | EnumDefinition
 
 InheritanceSpecifier = Identifier ( '(' Expression ( ',' Expression )* ')' )?
+
+StateVariableDecOrDef = TypeName ( 'public' | 'internal' | 'private' )? Identifier ('=' Expression)? ';'
```

Comment:
I think this should still be called declaration. An initial assigtnment is not a definition (because it is mutable).